src/csrc/training.cpp

```diff
@@ -129,6 +129,7 @@ void train_multiarg(const char* name, torchfort_tensor_list_t inputs_in, torchfo
 
   model->train();
   auto opt = models[name].optimizer.get();
```

Comment:
I think we can remove the .get() here and in the line below.

Reply:
Sure, I just removed `.get()` usage everywhere in this file.

Reply:
great, thanks. It is correct either way but I think it is cleaner using the smart pointers.